registry/storage/driver/s3-aws/s3.go

```diff
@@ -763,22 +763,24 @@ func (d *driver) Writer(ctx context.Context, path string, appendMode bool) (stor
 	return nil, storagedriver.PathNotFoundError{Path: path}
 }
 
-func (d *driver) statHead(ctx context.Context, s3Path string) (*storagedriver.FileInfoFields, error) {
+func (d *driver) statHead(ctx context.Context, path string) (*storagedriver.FileInfoFields, error) {
 	resp, err := d.S3.HeadObjectWithContext(ctx, &s3.HeadObjectInput{
 		Bucket: aws.String(d.Bucket),
-		Key:    aws.String(s3Path),
+		Key:    aws.String(d.s3Path(path)),
 	})
 	if err != nil {
 		return nil, err
 	}
 	return &storagedriver.FileInfoFields{
+		Path:    path,
 		IsDir:   false,
 		Size:    *resp.ContentLength,
 		ModTime: *resp.LastModified,
 	}, nil
 }
 
-func (d *driver) statList(ctx context.Context, s3Path string) (*storagedriver.FileInfoFields, error) {
+func (d *driver) statList(ctx context.Context, path string) (*storagedriver.FileInfoFields, error) {
+	s3Path := d.s3Path(path)
```

Comment:
the reason why this was done in `Stat` is so the `strings.Trim`ming done by `d.s3Path` is only ever done once regardless of any potential fall-through from head to list; this PR does it in some cases in both `statHead` and `listHead`. I really don't believe there's any need for that.

Reply:
But it's still done only once for each of those functions; `path` is never mutated ahead of time? Only because of that, creating `FileInfoFields` was not really atomic; the caller had to be aware that `Path` is not set, so has to be set manually.